variables.tf

```diff
@@ -282,10 +282,16 @@ variable "create_eks" {
   default     = true
 }
 
+variable "node_groups_defaults" {
+  description = "Map of values to be applied to all node groups. See `node_groups_defaults` variable in the `node_groups` submodule for more details"
+  type        = any
+  default     = {}
+}
+
 variable "node_groups" {
-  description = "A list of maps defining node group configurations to be defined using AWS EKS Managed Node Groups. See workers_group_defaults for valid keys."
+  description = "Map of map of node groups to create. See `node_groups` variable in the `node_groups` submodule for more details"
```

Comment:
Just thinking... Is it better to have a map or maps or a list of maps with a key `name` which will converted into a map during [expansion](https://github.com/terraform-aws-modules/terraform-aws-eks/pull/650/files#diff-52226e549f1fac8f550ec975e0bf901eR7) ? 

```hcl
  node_groups_expanded = { for k, v in var.node_groups : v.name => merge(
    var.node_groups_defaults,
    v,
  ) }
```

Reply:
I don't really see the advantage to this. It came up in discussions on #644. The output there was a list that had no relation to the order of the input list. This then makes life difficult for users if they want to associate the outputs to the items in their inputs.

Terraform will throw an error about duplicate keys if the name is specified twice. Not the most obvious error.

`for_each` would be much more useful if we weren't worrying about a `create_eks` variable or if Terraform worked better with Objects in ternary statements.

Reply:
Agree.